src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -953,6 +1076,12 @@ def enable_model_cpu_offload(self, gpu_id: Optional[int] = None, device: Union[t
                 The PyTorch device type of the accelerator that shall be used in inference. If not specified, it will
                 default to "cuda".
         """
+        is_pipeline_device_mapped = self.hf_device_map is not None and len(self.hf_device_map) > 1
```

Comment:
ohh how do you reset/disable the device_mapping  then?

Reply:
You need to load the pipeline again because the components of the pipeline would already be placed in their respective devices. Can add `reset_device_map()` that would get the pipeline in a state where all the components would be on the CPU so that methods like `enable_sequential_cpu_offload()` and `enable_model_cpu_offload()` work without hassle. Would that be nice?

Reply:
Added in f5d8a8f8.

Reply:
ok, so we would manually call `reset_device_map` if we want to apply the offloading? 
I think maybe we don't need a method then, since they can just do 
```
pipe.hf_device_map = None
pipe.enable_model_cpu_offload()
```

Reply:
That could be nice. But I don't want the users to rely on internal attributes that might change in the future (since this feature is a little experimental at the moment). 

So, I have also updated `hf_device_map` to be `_ hf_device_map` in fee86e96. 

WDYT?